pkgs/by-name/ll/llama-cpp/package.nix

```diff
@@ -0,0 +1,113 @@
+{ lib
+, cmake
+, darwin
+, fetchFromGitHub
+, stdenv
+, symlinkJoin
+
+, config
+, cudaSupport ? config.cudaSupport
+, cudaPackages ? { }
+
+, rocmSupport ? false
+, rocmPackages ? { }
+
+, openclSupport ? false
+, clblast
+}:
+
+let
+  cudatoolkit_joined = symlinkJoin {
+    name = "${cudaPackages.cudatoolkit.name}-merged";
+    paths = [
+      cudaPackages.cudatoolkit.lib
+      cudaPackages.cudatoolkit.out
+    ] ++ lib.optionals (lib.versionOlder cudaPackages.cudatoolkit.version "11") [
+      # for some reason some of the required libs are in the targets/x86_64-linux
+      # directory; not sure why but this works around it
+      "${cudaPackages.cudatoolkit}/targets/${stdenv.system}"
+    ];
+  };
```

Comment:
If i remember correctly, there is a push to move from cudatoolkit to cuda-redist. (for reference: `https://github.com/NixOS/nixpkgs/issues/232501`).

Not a blocker IMO

Reply:
Ok good to know - I'll monitor that issue.

Reply:
I think we can avoid using symlinkJoin since upstream uses `FindCUDAToolkit`.
see more details https://github.com/NixOS/nixpkgs/pull/235226#discussion_r1329126600

Reply:
Are you sure? As of 0.1.4 there was definitely the issue that a `lib` only path would be lookup up in the `out` path and fail (or the other way around, don't remember). I don't believe upstream switched their CMake setup in the meantime.

Reply:
https://github.com/NixOS/nixpkgs/issues/224291 seems related but is not closed and dead-looking.